crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractCustomResourceHandler.java

```diff
@@ -54,21 +60,53 @@ public void handle(CustomResourceInfo config) {
 
     ClassDependenciesVisitor traversedClassesVisitor = new ClassDependenciesVisitor(config.crClassName(), name);
 
-    TypeDefBuilder builder = new TypeDefBuilder(def);
-    if (config.specClassName().isPresent()) {
-      builder.accept(specReplicasPathDetector);
-    }
+    final ExecutorService executorService = Executors.newFixedThreadPool(
+      Math.min(5, Runtime.getRuntime().availableProcessors()));
```

Comment:
They are not going to compete for any resource because they are all independent tasks (that's what I've seen but I don't know this code very deeply). 

I haven't measured the cache hit ratio but the results show the parallel execution is drastically faster.